metaflow/plugins/argo/argo_workflows.py

```diff
@@ -2627,50 +2627,54 @@ def _heartbeat_daemon_template(self):
         )
         from kubernetes import client as kubernetes_sdk
 
-        return DaemonTemplate("heartbeat-daemon").container(
-            to_camelcase(
-                kubernetes_sdk.V1Container(
-                    name="main",
-                    # TODO: Make the image configurable
-                    image=resources["image"],
-                    command=cmds,
-                    env=[
-                        kubernetes_sdk.V1EnvVar(name=k, value=str(v))
-                        for k, v in env.items()
-                    ],
-                    env_from=[
-                        kubernetes_sdk.V1EnvFromSource(
-                            secret_ref=kubernetes_sdk.V1SecretEnvSource(
-                                name=str(k),
-                                # optional=True
+        return (
+            DaemonTemplate("heartbeat-daemon")
+            .service_account_name(resources["service_account"])
```

Comment:
do we need the service account for the hooks too?

Reply:
not sure, but they are also Templates so I am guessing the answer is yes. I can add this to the lifecycle hook templates if desired

Reply:
the existing hooks are only sending basic http requests with an argo built-in and not accessing any resources so hopefully they do not need the service account for now. we can revisit that in another PR if necessary.